tests/geometry/functional/test_so3.py

```diff
@@ -18,3 +18,13 @@ def test_exp(batch_size: int, dtype: torch.dtype):
     rng.manual_seed(0)
     tangent_vector = torch.rand(batch_size, 3, dtype=dtype, generator=rng)
     check_lie_group_function(so3, "exp", TEST_EPS, tangent_vector)
+
+
+@pytest.mark.parametrize("batch_size", [1, 20, 100])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+def test_adjoint(batch_size: int, dtype: torch.dtype):
+    rng = torch.Generator()
+    rng.manual_seed(0)
+    tangent_vector = torch.rand(batch_size, 3, dtype=dtype, generator=rng)
+    group = so3.exp(tangent_vector)
+    check_lie_group_function(so3, "adjoint", TEST_EPS, group)
```

Comment:
This logic is same as above. Could we instead just do
`@pytest.mark.parametrize("op", ["exp", "adjoint"])`

Reply:
Good catch! Let's definitely do this. You can call it `test_unary_operator`.

Reply:
Looking at inverse I realized there is an extra line there calling exp to get group from tangent. So not sure if we can do this for all unary ops.

Reply:
No, we can not do this for `adjoint` and `exp` since the inputs are different.

Reply:
We can add an if to handle ops that require group inputs, I think that would still be preferable. For example,
```python
if op_name in ["adjoint", ..., ]:
     op_input = so3.exp(tangent_vector)
else:
     op_input = tangent_vector
check_lie_group_function(so3, op_name, TEST_EPS, op_input)
```

Reply:
I will consider if doing this in the future since I'm planning to add more tests for `test_inverse` and `test_adjoint` which seem need specific tests like what we did in `theseus/geometry/lie_group`. But I created #377 for reminder.